azurerm/resource_arm_backup_protected_file_share_test.go

```diff
@@ -0,0 +1,315 @@
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %q", resourceName)
+		}
+
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for Azure Backup Protected File Share: %q", resourceName)
+		}
+
+		vaultName := rs.Primary.Attributes["recovery_vault_name"]
+		storageID := rs.Primary.Attributes["source_storage_account_id"]
+		fileShareName := rs.Primary.Attributes["source_file_share_name"]
+
+		parsedStorageID, err := azure.ParseAzureResourceID(storageID)
+		if err != nil {
+			return fmt.Errorf("[ERROR] Unable to parse source_storage_account_id '%s': %+v", storageID, err)
+		}
+		accountName, hasName := parsedStorageID.Path["storageAccounts"]
+		if !hasName {
+			return fmt.Errorf("[ERROR] parsed source_storage_account_id '%s' doesn't contain 'storageAccounts'", storageID)
+		}
+
+		protectedItemName := fmt.Sprintf("AzureFileShare;%s", fileShareName)
+		containerName := fmt.Sprintf("StorageContainer;storage;%s;%s", parsedStorageID.ResourceGroup, accountName)
+
+		client := testAccProvider.Meta().(*ArmClient).RecoveryServices.ProtectedItemsClient
+		ctx := testAccProvider.Meta().(*ArmClient).StopContext
+
+		resp, err := client.Get(ctx, vaultName, resourceGroup, "Azure", containerName, protectedItemName, "")
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Azure Backup Protected File Share %q (resource group: %q) was not found: %+v", protectedItemName, resourceGroup, err)
+			}
+
+			return fmt.Errorf("Bad: Get on recoveryServicesVaultsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMBackupProtectedFileShare_base(rInt int, location string) string {
+	rstr := strconv.Itoa(rInt)
+	return fmt.Sprintf(` 
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctest%[3]s"
+  location                 = "${azurerm_resource_group.test.location}"
+  resource_group_name      = "${azurerm_resource_group.test.name}"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_share" "test" {
+  name                 = "acctest-%[1]d"
+  storage_account_name = "${azurerm_storage_account.test.name}"
+  metadata             = {}
+  
+  lifecycle {
+	ignore_changes = [metadata] // Ignore changes Azure Backup makes to the metadata
+  }
+}
+
+resource "azurerm_recovery_services_vault" "test" {
+  name                = "acctest-%[1]d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "Standard"
+}
+
+resource "azurerm_backup_policy_file_share" "test1" {
+  name                = "acctest-%[1]d"
```

Comment:
Could we put some capitals in here to check the API handles them correctly?
```suggestion
  name                = "acctest-PFS-%[1]d"
```